File: core/src/main/java/com/lomekwi/cave/ui/editpanel/settings/EntryTree.java
package com.lomekwi.cave.ui.editpanel.settings;

import com.kotcrab.vis.ui.widget.VisTree;
import com.lomekwi.cave.ui.listeners.ChangeListenerX;

public class EntryTree extends VisTree<EntryNode, EntryTable> {
    public EntryTree(SettingsTable settingsTable) {
        addListener(new ChangeListenerX(() -> {
            if (!getSelection().isEmpty()) {
                settingsTable.onEntrySelected(getSelection().first().getValue());
            }
        }));
    }
}

File: core/src/main/scala/com/lomekwi/cave/ui/settings/EntryTree.scala
package com.lomekwi.cave.ui.settings

import com.kotcrab.vis.ui.widget.VisTree
import com.lomekwi.cave.ui.listeners.ChangeListenerX

class EntryTree(settingsTable: SettingsTable) extends VisTree[EntryNode, EntryTable] {
  addListener(new ChangeListenerX(() => {
    if (!getSelection.isEmpty) {
      settingsTable.onEntrySelected(getSelection.first().getValue)
    }
  }))
}
